apps/poc-state-plugin/src/plugin.ts

```diff
@@ -91,9 +121,14 @@ penpot.on('selectionchange', () => {
   selection.forEach((shape) => {
     console.log(`== ${shape.name} ==`);
     console.log(`Tokens: ${JSON.stringify(shape.tokens)}`);
-    console.log(`Variant props: ${JSON.stringify(shape.component()?.variantProps)}`);
+    const component = shape.component();
+    if (component?.isVariant()) {
```

Comment:
If you use the utility method `penpot.utils.types.isVariantComponent(component)` then the casting later won't be necessary.

Reply:
Fixed